apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1.IntegrationTests/DatastoreDbTest.cs

```diff
@@ -120,6 +121,46 @@ public void RunQuery_NoResults()
             Assert.Equal(MoreResultsType.NoMoreResults, singleResponse.Batch.MoreResults);
         }
 
+        [Fact]
+        public void InFilterQuery()
+        {
+            var db = _fixture.CreateDatastoreDb();
+            var keyFactory = db.CreateKeyFactory("inFilter");
+
+            var actualQuery = new Query("inFilter") { Filter = Filter.In("1", null) };
+            var expected = new Filter
+            {
+                PropertyFilter = new PropertyFilter
+                {
+                    Op = Operator.In,
+                    Property = new PropertyReference { Name = "1" },
+                    Value = Value.ForNull()
+                }
+            };
+            var expectedQuery = new Query("inFilter") { Filter = expected };
+            Assert.Equal(db.RunQuery(actualQuery), db.RunQuery(expectedQuery));
```

Comment:
This still doesn't tell us what the server actually does though. That's what I really want to know, so we can document it - or prohibit it if it's not useful.

We want to know what it will match, if anything. This is effectively checking that the expected query is the actual query, and we can do that in a unit test. It's the *impact* of doing that in terms of returned results that's important.